java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/menu/handlers/CreateItemHandler.java

```diff
@@ -22,18 +22,17 @@
 @Component
 public class CreateItemHandler extends MenuBaseCommandHandler<CreateItemCommand> {
 
-  private UUID itemId;
-
   public CreateItemHandler(
       MenuRepository menuRepository, ApplicationEventPublisher applicationEventPublisher) {
     super(menuRepository, applicationEventPublisher);
   }
 
   @Override
   Optional<UUID> handleCommand(Menu menu, CreateItemCommand command) {
-    itemId = UUID.randomUUID();
-    Category category = addItem(getCategory(menu, command), command);
+    UUID itemId = UUID.randomUUID();
+    Category category = addItem(getCategory(menu, command), command, itemId);
```

Comment:
We should be able pass the command will not need to send the item id since item id will be set before command variable is passed in this function